components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+					tonumber(opponent.sumscore) or -1
+				if opponent.score > bestof / 2 then
+					match.finished = Logic.emptyOr(match.finished, true)
+					match.winner = tonumber(match.winner or '') or opponentIndex
+				end
+			end
+
+			if Logic.readBool(match.cancelled) then
+				match.finished = true
+				if String.isEmpty(match.resulttype) and String.isEmpty(opponent.score) then
+					match.resulttype = 'np'
+					opponent.score = opponent.score or -1
+				end
+			end
+		else
+			break
+		end
+	end
+
+	CustomMatchGroupInput._determineWinnerIfMissing(match)
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+		if match.winner == 'draw' or tonumber(match.winner) == 0 or
+				(match.opponent1.score == bestof / 2 and match.opponent1.score == match.opponent2.score) then
+			match.finished = true
+			match.winner = 0
+			match.resulttype = 'draw'
+		end
+		if tonumber(match.winner) == opponentIndex or match.resulttype == 'draw' then
+			opponent.placement = 1
+		elseif Logic.isNumeric(match.winner) then
+			opponent.placement = 2
+		end
+	end
+
+	return match
+end
+
+function CustomMatchGroupInput._determineWinnerIfMissing(match)
+	if Logic.readBool(match.finished) and not match.winner then
+		local scores = Array.mapIndexes(function(opponentIndex)
+			local opponent = match['opponent' .. opponentIndex]
+			if not opponent then
+				return nil
+			end
+			return match['opponent' .. opponentIndex].score or -1 end
+		)
+		local maxScore = math.max(unpack(scores) or 0)
+		-- if we have a positive score and the match is finished we also have a winner
+		if maxScore > 0 then
+			local maxIndexFound = false
+			for opponentIndex, score in pairs(scores) do
+				if maxIndexFound and score == maxScore then
+					match.winner = 0
+					break
+				elseif score == maxScore then
+					maxIndexFound = true
+					match.winner = opponentIndex
+				end
+			end
+		end
+	end
+
+	return match
+end
+
+--[[
+
+OpponentInput functions
+
+]]--
+function CustomMatchGroupInput._opponentInput(match)
+	local opponentIndex = 1
+	local opponent = match['opponent' .. opponentIndex]
+
+	while opponentIndex <= MAX_NUM_OPPONENTS and Logic.isNotEmpty(opponent) do
+		opponent = Json.parseIfString(opponent) or Opponent.blank()
+
+		-- Convert byes to literals
+		if
+			string.lower(opponent.template or '') == 'bye'
+			or string.lower(opponent.name or '') == 'bye'
+		then
+			opponent = {type = Opponent.literal, name = 'BYE'}
+		end
+
+		--process input
+		if opponent.type == Opponent.team or opponent.type == Opponent.solo or
+			opponent.type == Opponent.duo or opponent.type == Opponent.literal then
+
+			opponent = CustomMatchGroupInput.processOpponent(opponent, match.timestamp)
+		else
+			error('Unsupported Opponent Type')
+		end
+
+		-- Retrieve icon for team
+		if opponent.type == Opponent.team then
+			opponent.icon, opponent.icondark = CustomMatchGroupInput.getIcon(opponent.template)
+			match.hasTeamOpponent = true
+		end
```

Comment:
Move to processOpponent

Reply:
icon stuff moved there
`match.hasTeamOpponent` since match is not available there

Reply:
Hopefully we don't need hasTeamOpponent variable :)